app/models/patient.rb

```diff
@@ -25,23 +25,20 @@ class Patient < ApplicationRecord
 
   validate :valid_birth_date
 
-  after_initialize :set_main_ubs
+  # Only set new main_ubs if it is empty or there was a change to the neighborhood
+  before_validation :set_main_ubs!, if: proc { |r| r.neighborhood_changed? || r.main_ubs_id.blank? }
 
   scope :bedridden, -> { where(bedridden: true) }
 
   # TODO: remove `chronic` field from schema
   enum target_audience: [:kid, :elderly, :chronic, :disabled, :pregnant, :postpartum, :teacher, :over_55, :without_target]
 
   def current_appointment
-    appointments.order(:start).select(&:active?).last
```

Comment:
`select(&:active?)` is pure Ruby and makes Rails call all appointments and iterate through each one. Replacing with `active` scope will instead use ARel and solve the issue on a single SQL query on the DB, thus saving time.